components/apps_form/apps_form_field/apps_form_select_field.tsx

```diff
@@ -73,6 +89,26 @@ export default class AppsFormSelectField extends React.PureComponent<Props, Stat
         return this.props.performLookup(this.props.field.name, userInput);
     }
 
+    loadDynamicUserOptions = async (userInput: string): Promise<AppSelectOption[]> => {
+        const usersSearchResults: UserAutocomplete = await this.props.actions.autocompleteUsers(userInput.toLowerCase());
+
+        return usersSearchResults.users.map((user) => {
+            const label = this.props.teammateNameDisplay ? displayUsername(user, this.props.teammateNameDisplay) : user.username;
+
+            return {...user, label, value: user.id, icon_data: imageURLForUser(user.id)};
+        });
+    }
+
+    loadDynamicChannelOptions = async (userInput: string): Promise<AppSelectOption[]> => {
+        let channelsSearchResults: Channel[] = [];
+
+        await this.props.actions.autocompleteChannels(userInput.toLowerCase(), (data) => {
+            channelsSearchResults = data;
+        }, () => {});
+
+        return channelsSearchResults.map((channel) => ({...channel, label: channel.name, value: channel.id}));
```

Comment:
I'm thinking we should use the channel's display name here

```suggestion
        return channelsSearchResults.map((channel) => ({...channel, label: channel.display_name, value: channel.id}));
```